### PR TITLE
Jashton/prepare for dangerous property analyzer

### PR DESCRIPTION
**Pull request by JeffAshton**

### CODE REVIEW

File: src/D2L.CodeStyle.Analyzers/ApiUsage/DangerousMemberUsages/DangerousMemberUsagesAnalyzer.cs [hunk 1]
```diff
@@ -63,39 +63,54 @@ IImmutableSet<ISymbol> dangerousMethods
...
 			}
 
-			ReportDiagnostic( context, methodSymbol );
+			return true;
 		}
 
-		private static bool IsDangerousMethodSymbol(
-				ISymbol methodSymbol,
-				IImmutableSet<ISymbol> dangerousMethods
+		private static bool IsDangerousMemberSymbol(
```
Comment (JeffAshton) [inline src/D2L.CodeStyle.Analyzers/ApiUsage/DangerousMemberUsages/DangerousMemberUsagesAnalyzer.cs]: Broke this method out of the existing ```AnalyzeInnovation```, but does the same thing

File: src/D2L.CodeStyle.Analyzers/ApiUsage/DangerousMemberUsages/DangerousMemberUsagesAnalyzer.cs [hunk 2]
```diff
@@ -63,39 +63,54 @@ IImmutableSet<ISymbol> dangerousMethods
...
+			if( !AnalyzePotentiallyDangerousMember( context, methodSymbol, auditedAttributeType, unauditedAttributeType, dangerousMethods ) ) {
 				return;
 			}
 
-			if( !IsDangerousMethodSymbol( methodSymbol, dangerousMethods ) ) {
-				return;
+			ReportDiagnostic( context, methodSymbol, Diagnostics.DangerousMethodsShouldBeAvoided );
+		}
+
+		private bool AnalyzePotentiallyDangerousMember(
```
Comment (JeffAshton) [inline src/D2L.CodeStyle.Analyzers/ApiUsage/DangerousMemberUsages/DangerousMemberUsagesAnalyzer.cs]: Broke this method out of the existing ```AnalyzeInnovation```, but does the same thing

File: src/D2L.CodeStyle.Analyzers/ApiUsage/DangerousMemberUsages/DangerousMemberUsagesAnalyzer.cs [hunk 3]
```diff
@@ -63,39 +63,54 @@ IImmutableSet<ISymbol> dangerousMethods
...
+				.Any( attr => IsAuditedAttribute( auditedAttributeType, unauditedAttributeType, attr, memberSymbol ) );
 
 			if( isAudited ) {
-				return;
+				return false;
 			}
 
-			ReportDiagnostic( context, methodSymbol );
+			return true;
 		}
```
Comment (JeffAshton) [inline src/D2L.CodeStyle.Analyzers/ApiUsage/DangerousMemberUsages/DangerousMemberUsagesAnalyzer.cs]: Renames from ```method``` to ```member```

File: src/D2L.CodeStyle.Analyzers/ApiUsage/DangerousMemberUsages/DangerousMemberUsagesAnalyzer.cs [hunk 4]
```diff
@@ -12,8 +12,8 @@ namespace D2L.CodeStyle.Analyzers.ApiUsage.DangerousMemberUsages {
 	[DiagnosticAnalyzer( LanguageNames.CSharp )]
 	internal sealed class DangerousMemberUsagesAnalyzer : DiagnosticAnalyzer {
 
-		private const string AuditedAttributeFullName = "D2L.CodeStyle.Annotations.DangerousMethodUsage+AuditedAttribute";
-		private const string UnauditedAttributeFullName = "D2L.CodeStyle.Annotations.DangerousMethodUsage+UnauditedAttribute";
+		private const string DangerousMethodAuditedAttributeFullName = "D2L.CodeStyle.Annotations.DangerousMethodUsage+AuditedAttribute";
```
Comment (travisMiehm) [inline src/D2L.CodeStyle.Analyzers/ApiUsage/DangerousMemberUsages/DangerousMemberUsagesAnalyzer.cs]: personally, I think this rename is overkill.  The string is `private`, and we're in a class named Dangerous*, so adding that on is redundant.  Unless you're going to be making this not private

File: src/D2L.CodeStyle.Analyzers/ApiUsage/DangerousMemberUsages/DangerousMemberUsagesAnalyzer.cs (hunk 2 above)
Comment (travisMiehm) [reply src/D2L.CodeStyle.Analyzers/ApiUsage/DangerousMemberUsages/DangerousMemberUsagesAnalyzer.cs]: Please tell me you'll fix that typo as part of this? `Invocation`

Comment (JeffAshton) [reply src/D2L.CodeStyle.Analyzers/ApiUsage/DangerousMemberUsages/DangerousMemberUsagesAnalyzer.cs]: I can fix typeos after

**Pull request merged**